chainer/testing/math_function_test.py

```diff
@@ -0,0 +1,207 @@
+import unittest
+
+import numpy
+
+import chainer
+from chainer import cuda
+import chainer.functions as F
+from chainer.testing import attr
+from chainer.testing import condition
+
+
+def func_class(func):
+    name = func.__name__
+    name = name[0].upper() + name[1:]
+    return getattr(F, name, None)
+
+
+def make_data_default(dtype, shape):
+    x = numpy.random.uniform(-1, 1, shape).astype(dtype)
+    gy = numpy.random.uniform(-1, 1, shape).astype(dtype)
+    return x, gy
+
+
+def math_function_test(func, func_expected=None, label_expected=None,
+                       make_data=None):
+    """Decorator to test Chainer functions lifting mathematical numpy functions.
+
+    This decorator is for testing Chainer functions lifted from corresponding
+    mathematical numpy and cupy functions, and optionally ones composed with
+    such other Chainer functions. Forward and backward computations on CPU and
+    GPU across parameterized ``dtype`` and ``shape`` are tested.
+
+    Args:
+        func(~chainer.Function): Chainer function to be tested by
```

Comment:
Alternatively, directory pass Chainer Functions to this decorator as `func` argument and do not use `func_class`.